### PR TITLE
feat: validate onchain space on follows action

### PR DESCRIPTION
**Pull request by wa0x6e**
This PR:

- moves the space validation for follow/unfollow actions to the writer (for custom validation)
- connect to sx api to validate space from sx

### How to test

- Boot a local instance of sx-monorepo, using this PR branch
- Follow and unfollow some spaces, it should work
- Boot a local instance of snapshot, using this PR branch 
- Follow and unfollow some spaces, it should work

### CODE REVIEW

File: src/ingestor.ts [hunk 1]
```diff
@@ -62,10 +62,14 @@ export default async function ingestor(req) {
...
-      if (!space) return Promise.reject('unknown space');
-      network = space.network;
-      if (space.voting?.aliased) aliased = true;
+
+      // Disable space checking for follow/unfollow, checking delegated to the writer
+      if (!['follow', 'unfollow'].includes(type)) {
+        const space = await getSpace(message.space);
+        if (!space) return Promise.reject('unknown space');
+        network = space.network;
+        if (space.voting?.aliased) aliased = true;
```
Comment (ChaituVR) [inline src/ingestor.ts:71]: I think this line should be outside `if` if not, aliased will `false` for `follow` and `unfollow`

File: src/ingestor.ts [hunk 2]
```diff
@@ -62,10 +62,14 @@ export default async function ingestor(req) {
...
-      const space = await getSpace(message.space);
-      if (!space) return Promise.reject('unknown space');
-      network = space.network;
-      if (space.voting?.aliased) aliased = true;
+
+      // Disable space checking for follow/unfollow, checking delegated to the writer
+      if (!['follow', 'unfollow'].includes(type)) {
+        const space = await getSpace(message.space);
+        if (!space) return Promise.reject('unknown space');
+        network = space.network;
```
Comment (ChaituVR) [inline src/ingestor.ts:70]: maybe this line also?

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/ingestor.ts (hunk 1 above)
Comment (wa0x6e) [reply src/ingestor.ts:71]: But outside the if, `space` is undefined, so aliased will always be false anyway

File: src/ingestor.ts [hunk 3]
```diff
@@ -62,10 +62,14 @@ export default async function ingestor(req) {
     let aliased = false;
     if (!['settings', 'alias', 'profile'].includes(type)) {
       if (!message.space) return Promise.reject('unknown space');
-      const space = await getSpace(message.space);
-      if (!space) return Promise.reject('unknown space');
-      network = space.network;
-      if (space.voting?.aliased) aliased = true;
+
+      // Disable space checking for follow/unfollow, checking delegated to the writer
+      if (!['follow', 'unfollow'].includes(type)) {
```
Comment (ChaituVR) [inline src/ingestor.ts:67]: I think we can include `follow` and `unfollow` in line `63` no?

Comment (wa0x6e) [reply src/ingestor.ts:67]: will lose the `message.space` params existence checking

**Comment by wa0x6e** [comment]
PR refactor to include the SX spaces validation

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/writer/follow.ts [hunk 4]
```diff
@@ -27,6 +28,14 @@ export async function verify(message): Promise<any> {
...
   }
 
+  if (!message.network || message.network === defaultNetwork) {
+    const space = await getSpace(message.space);
+    if (!space) return Promise.reject('unknown space');
+  } else {
+    const spaceExist = await sxSpaceExists(message.space);
+    if (!spaceExist) return Promise.reject('unknown space');
+  }
+
```
Comment (ChaituVR) [inline src/writer/follow.ts:38]: Same here

File: src/helpers/actions.ts [hunk 5]
```diff
@@ -37,7 +39,16 @@ export async function getProposal(space, id) {
...
 }
 
-export async function getSpace(id: string, includeDeleted = false) {
+export async function getSpace(id: string, includeDeleted = false, network = defaultNetwork) {
+  if (network !== defaultNetwork) {
+    const spaceExist = await sxSpaceExists(id);
+    if (!spaceExist) return false;
+
+    return {
+      network: 1
```
Comment (ChaituVR) [inline src/helpers/actions.ts:48]: feel like this may trouble us in the future. 😢

Comment (wa0x6e) [reply src/helpers/actions.ts:48]: We could pass `0` instead

Comment (ChaituVR) [reply src/helpers/actions.ts:48]: hmm if it works then yea 😄

**Review by ChaituVR — APPROVED** [review]
utAck

**Pull request merged**